base/array.hpp

```diff
@@ -102,16 +130,33 @@ class BoundedArray final {
 using Bytes = Array<std::uint8_t>;
```

Comment:
Are these names still useful?  I mean, we don't have them for `char` and `char16_t`.

Reply:
We could have them, if we can find good names (Array<char> is some sort of mutable string view, Array<char const> is a string view by any other name, and then there's the same in UTF-16).